pkg/armhelpers/compute.go

```diff
@@ -36,13 +36,15 @@ func (az *AzureClient) DeleteVirtualMachine(ctx context.Context, resourceGroup,
 }
 
 // ListVirtualMachineScaleSets returns (the first page of) the vmss resources in the specified resource group.
-func (az *AzureClient) ListVirtualMachineScaleSets(ctx context.Context, resourceGroup string) (compute.VirtualMachineScaleSetListResultPage, error) {
-	return az.virtualMachineScaleSetsClient.List(ctx, resourceGroup)
+func (az *AzureClient) ListVirtualMachineScaleSets(ctx context.Context, resourceGroup string) (VirtualMachineScaleSetListResultPage, error) {
```

Comment:
FYI this will require a change to AKS implementation, as it consumes this client+method.

Reply:
It shouldn't cause a problem as the interface is 1-1 with the compute type, and all other methods currently depend on these interfaces. If we're lucky, the import will already be present.
Otherwise, good reminder to always depend on our interface types and never on the compute types...